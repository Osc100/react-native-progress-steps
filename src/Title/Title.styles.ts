import { DEFAULT_ORIENTATION } from '../constants';
import { StyleSheet } from 'react-native';
import type { Tstyles } from './Title.types';

const styles: Tstyles = {
  vertical: {
    text: {
      paddingTop: 12,
      paddingBottom: 12,
      paddingLeft: 5,
      fontWeight: 'bold',
    },
  },
  horizontal: {
    text: {
      padding: 5,
      textAlign: 'center',
      fontWeight: 'bold',
    },
  },
};

const useStyles = (orientation = DEFAULT_ORIENTATION) =>
  StyleSheet.create(styles[orientation]);

export default useStyles;
