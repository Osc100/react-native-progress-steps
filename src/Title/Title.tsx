import { Animated } from 'react-native';
import { EASING } from '../constants';
import React, { FC, memo, useEffect, useRef } from 'react';
import styles from './Title.styles';
import type { ITitle } from './Title.types';

const Title: FC<ITitle> = ({
  children,
  stepState: { isActive, isCompleted, isFirstInteraction } = {},
}) => {
  const animation = useRef(new Animated.Value(0)).current;

  useEffect(() => {
    Animated.timing(animation, {
      toValue: isActive || isCompleted ? 1 : 0,
      duration: 400,
      easing: EASING,
      useNativeDriver: false,
    }).start();
  }, [isActive, isCompleted]); // eslint-disable-line react-hooks/exhaustive-deps

  const getColor = React.useCallback(() => {
    if (isFirstInteraction) {
      if (isCompleted || isActive) {
        return 'black';
      }

      return 'gray';
    }

    if (isCompleted) {
      return 'black';
    }

    return animation.interpolate({
      inputRange: [0, 1],
      outputRange: ['gray', 'black'],
    });
  }, [isFirstInteraction, isCompleted, isActive]); // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <Animated.Text style={[styles.text, { color: getColor() }]}>
      {children}
    </Animated.Text>
  );
};

export default memo(
  Title,
  (prevProps, nextProps) =>
    prevProps.stepState?.isActive === nextProps.stepState?.isActive &&
    prevProps.stepState?.isCompleted === nextProps.stepState?.isCompleted
);
