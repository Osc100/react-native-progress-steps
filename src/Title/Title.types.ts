import type { IstepState } from '../ProgressSteps.types';

export interface ITitle {
  children: string;
  stepState?: IstepState;
}
