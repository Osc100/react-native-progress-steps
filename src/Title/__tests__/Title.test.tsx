import React from 'react';
import Title from '../';
import { render } from '@testing-library/react-native';

const stepState = {
  index: 0,
  isActive: true,
  isCompleted: false,
  isFirst: true,
  isLast: false,
  isFirstInteraction: true,
};

describe('Title', () => {
  it('should render Title', () => {
    const { queryByText, toJSON } = render(
      <Title stepState={stepState}>Hello title</Title>
    );

    expect(queryByText('Hello title')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
});
