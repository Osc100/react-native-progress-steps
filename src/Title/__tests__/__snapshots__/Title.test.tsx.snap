// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should render Title 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "black",
        "fontWeight": "bold",
        "paddingBottom": 12,
        "paddingLeft": 5,
        "paddingTop": 12,
      }
    }
  >
    Hello title
  </Text>
</View>
`;
